CRM/Event/Form/Registration.php

```diff
@@ -2125,4 +2125,19 @@ protected function isMaxValueValidationRequired(): bool {
     return FALSE;
   }
 
+  /**
+   * Is this event configured to show the payment processors on the confirmation form?
+   *
+   * @return bool
+   */
+  protected function isShowPaymentOnConfirm(): bool {
+    $showPaymentOnConfirm = (
+      in_array($this->getEventID(), \Civi::settings()->get('event_show_payment_on_confirm')) ||
+      in_array('all', \Civi::settings()->get('event_show_payment_on_confirm')) ||
+      in_array('multiparticipant', \Civi::settings()->get('event_show_payment_on_confirm')) && $this->getEventValue('is_multiple_registrations')
```

Comment:
living on the edge here without safety brackets :) ok.

Reply:
Think it's better with.. added!